### PR TITLE
feat: add vacuum table dry run result table

### LINKED ISSUE #14810: feat: vacuum ... dry run returns file size
**Summary**

The Databend EE vacuum dry run currently only shows file names. It should also show file sizes, so users can see how much data will be cleaned up.

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat: add vacuum table dry run result table

- Fixes #14810

add `vacuum table dry run summary` grammar:
* when dry run without summary option, output column will be `file` and `file_size`;
* when dry run with summary option, output column will be `total_files` and `total_size`.

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [x] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14851)

### CODE REVIEW

File: src/query/ast/src/ast/statements/table.rs [hunk 1]
```diff
@@ -698,13 +698,17 @@ pub enum CompactTarget {
 #[derive(Debug, Clone, PartialEq, Drive, DriveMut)]
 pub struct VacuumTableOption {
     #[drive(skip)]
-    pub dry_run: bool,
+    // Some(true) means summary
+    pub dry_run: Option<bool>,
```
Comment (BohuTANG) [inline src/query/ast/src/ast/statements/table.rs:702]: Seems we need a new filed: `pub dry_run_summary: bool`

File: src/query/service/src/interpreters/interpreter_table_vacuum.rs [hunk 2]
```diff
@@ -161,9 +166,25 @@ impl Interpreter for VacuumTableInterpreter {
...
-                DataBlock::new_from_columns(vec![StringType::from_data(purge_files)]),
-            ]),
+            Some(purge_files) => {
+                let mut file_sizes = vec![];
+                let operator = fuse_table.get_operator();
+                for file in &purge_files {
+                    file_sizes.push(operator.stat(file).await?.content_length());
+                }
+
+                if self.plan.option.dry_run.unwrap() {
```
Comment (BohuTANG) [inline src/query/service/src/interpreters/interpreter_table_vacuum.rs:176]:
```
if self.plan.option.dry_run_summary ...
```

File: src/query/ast/src/ast/statements/table.rs (hunk 1 above)
Comment (BohuTANG) [reply src/query/ast/src/ast/statements/table.rs:702]: The syntax is easy to understand:
`VACUUM ... [DRY RUN] | [DRY RUN SUMMARY]`

than:
`VACUUM ... [DRY RUN] [SUMMARY]`

Comment (lichuang) [reply src/query/ast/src/ast/statements/table.rs:702]: if we have `dry_run_summary` option, it means that is option is independent with `dry_run` option, but it is not an independent.so:

* dry_run.is_none(): not dry run
*  dry_run.is_some(summary): dry run with summary option.

Comment (BohuTANG) [reply src/query/ast/src/ast/statements/table.rs:702]: OK.
After the explain it's make sense now, better  to add the above comments to the code to make it more understandable.

File: src/query/service/src/interpreters/interpreter_table_vacuum.rs (hunk 2 above)
Comment (BohuTANG) [reply src/query/service/src/interpreters/interpreter_table_vacuum.rs:177]: `unwrap` is a not a good idea as a if condition.

**Review by BohuTANG — APPROVED** [review]
LGTM

**Pull request merged**